Sources/flintc/Compiler.swift

```diff
@@ -92,9 +165,14 @@ struct Compiler {
     print("Failed to compile.")
     exit(1)
   }
+
+  func exitWithCompilerCreationError() -> Never {
+    print("The compiler has not been initialised in a proper way.")
```

Comment:
This is far from Swifty. You should not be able to initialise the object incorrectly and you do so via non-nullable properties and delegating constructors. And if that isn’t possible you should probably employ the builder pattern or something similar.